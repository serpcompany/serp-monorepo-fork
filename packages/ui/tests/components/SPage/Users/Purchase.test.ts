import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import Purchase from '../../../../components/SPage/Users/Purchase.vue';
import ComponentRender from '../../../componentRender';

// Route mock so we can simulate different query parameters
let routeMock_: { query: Record<string, unknown> } = { query: {} };

mockNuxtImport('useRoute', () => () => routeMock_);
mockNuxtImport('useRouter', () => () => ({
  push: () => {},
  replace: () => {},
  resolve: () => {
    return { href: '' };
  }
}));

describe('SPage/Users/Purchase Snapshot', () => {
  const scenarios: [string, { query: Record<string, unknown> }][] = [
    [
      'renders success state correctly',
      { query: { success: 'true', redirectTo: '/success' } }
    ],
    [
      'renders cancelled state correctly',
      { query: { cancel: 'true', redirectTo: '/cancel' } }
    ],
    [
      'renders error state correctly',
      { query: { error: 'Payment failed', redirectTo: '/error' } }
    ],
    ['renders unknown state correctly', { query: {} }]
  ];

  it.each(scenarios)('%s', async (desc, { query }) => {
    // Update our route mock for the current scenario
    routeMock_ = { query };

    const html = await ComponentRender(
      `SPage/Users/Purchase ${desc}`,
      {},
      Purchase
    );
    expect(html).toMatchSnapshot();
  });
});
