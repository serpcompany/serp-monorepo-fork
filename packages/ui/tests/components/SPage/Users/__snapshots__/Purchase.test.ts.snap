// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPage/Users/Purchase Snapshot > renders cancelled state correctly 1`] = `
"<div class="success-container">
  <div>
    <h1>❌ Cancelled</h1>
    <p>Your payment was cancelled.</p>
  </div>
  <p> Redirecting to <a href="#">/cancel</a> in 10 seconds. </p>
</div>"
`;

exports[`SPage/Users/Purchase Snapshot > renders error state correctly 1`] = `
"<div class="success-container">
  <div>
    <h1>⚠️ Error</h1>
    <p>There was an error processing your payment. - Payment failed</p>
  </div>
  <p> Redirecting to <a href="#">/error</a> in 10 seconds. </p>
</div>"
`;

exports[`SPage/Users/Purchase Snapshot > renders success state correctly 1`] = `
"<div class="success-container">
  <div>
    <h1>🎉 Success!</h1>
    <p>Your payment was processed successfully.</p>
  </div>
  <p> Redirecting to <a href="#">/success</a> in 10 seconds. </p>
</div>"
`;

exports[`SPage/Users/Purchase Snapshot > renders unknown state correctly 1`] = `
"<div class="success-container">
  <div>
    <h1>❓ Unknown</h1>
    <p>Unknown error.</p>
  </div>
  <p> Redirecting to <a href="#">/</a> in 10 seconds. </p>
</div>"
`;
